runtime/src/main/java/io/quarkus/reactivemessaging/http/runtime/AbstractSink.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.reactivemessaging.http.runtime;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.Flow;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.jboss.logging.Logger;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.groups.UniRetry;
+import io.smallrye.reactive.messaging.providers.helpers.MultiUtils;
+import io.smallrye.reactive.messaging.providers.helpers.SenderProcessor;
+
+abstract class AbstractSink {
+
+    private final Flow.Subscriber<? extends Message<?>> subscriber;
+
+    public AbstractSink(Logger log, String url,
+            int maxRetries, double jitter, Optional<Duration> delay,
+            long inflights, boolean waitForCompletion) {
+        if (inflights <= 0) {
+            throw new IllegalArgumentException("Inflights must be greater than 0, but was " + inflights);
+        }
+        SenderProcessor processor = new SenderProcessor(inflights, waitForCompletion, m -> {
+            Uni<Void> send = send(m);
+
+            log.debugf("maxRetries: %d for %s", maxRetries, url);
+            if (maxRetries > 0) {
+                UniRetry<Void> retry = send.onFailure().retry();
+                if (delay.isPresent()) {
+                    retry = retry.withBackOff(delay.get()).withJitter(jitter);
+                }
+                send = retry.atMost(maxRetries);
+            }
+
+            return send
+                    .onItemOrFailure().transformToUni((result, error) -> {
+                        if (error != null) {
+                            return Uni.createFrom().completionStage(
+                                    m.nack(error).thenRun(() -> log.debugf(error, "Error responding to %s", url)));
```

Comment:
I'm wondering if log level should be warn here and in line 48 but this may case a lot of logs produced in case of corresponding endpoint down.